product_total_weight_from_packaging/models/product.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import models
+
+
+class ProductProduct(models.Model):
+
+    _inherit = "product.product"
+
+    def get_total_weight_from_packaging(self, qty):
+        self.ensure_one()
+        qty_by_packaging_with_weight = self.with_context(
+            **{
+                "_packaging_filter": lambda p: p.max_weight,
+                "_with_packaging_weight": True,
+            }
+        ).product_qty_by_packaging(qty)
+        total_weight = 0
+        for packaging_dict in qty_by_packaging_with_weight:
+            if packaging_dict.get("is_unit"):
+                pack_weight = self.weight
```

Comment:
is this needed? You set the weight for units on line 30 as `self.weight` as well. :thinking: